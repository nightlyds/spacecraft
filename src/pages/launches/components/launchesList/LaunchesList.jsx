import React from "react";
import PropsTypes from "prop-types";
import Success from "../../images/success.svg";
import Fail from "../../images/fail.svg";

const LaunchesList = ({ launches, oneLaunchOpen }) => {
  const api_request = launches ? launches.data : [];
  return (
    <div className="launches-list">
      {api_request &&
        api_request.map((point, index) => (
          <div className="launches-list-point" key={`${point}_${index}`} onClick={() => {oneLaunchOpen(point.flight_number)}}>
            <div className="launches-list-point-box">
              <span className="launches-list-point-detail">
                {point.mission_name}
              </span>
            </div>
            <div className="launches-list-point-box">
              <span className="launches-list-point-detail">
                {point.launch_date_utc}
              </span>
            </div>
            <div className="launches-list-point-box">
              {point.launch_success === true ? (
                <img
                  src={Success}
                  alt="success"
                  className="launches-list-success-icon"
                />
              ) : (
                <img
                  src={Fail}
                  alt="fail"
                  className="launches-list-success-icon"
                />
              )}
            </div>
            <div className="launches-list-point-box">
              <span className="launches-list-point-detail">
                {point.upcoming === true ? 'yes' : 'no'}
              </span>
            </div>
          </div>
        ))}
    </div>
  );
};

LaunchesList.propTypes = {
  launches: PropsTypes.object,
  oneLaunchOpen: PropsTypes.func
};

export default LaunchesList;
