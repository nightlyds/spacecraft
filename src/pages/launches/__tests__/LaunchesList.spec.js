import React from "react";
import LaunchesList from "../components/launchesList/LaunchesList.jsx";
import { shallow } from "enzyme";

const setUp = (props) => shallow(<LaunchesList {...props} />);

describe("LaunchesList component tests", () => {
  describe("Snapshot", () => {
    let component;

    beforeEach(() => {
      component = setUp();
    });

    it("LaunchesList default snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
