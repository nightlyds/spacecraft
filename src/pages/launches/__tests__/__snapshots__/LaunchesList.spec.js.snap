// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchesList component tests Snapshot LaunchesList default snapshot 1`] = `
<div
  className="launches-list"
/>
`;
